### PR TITLE
Added support for application specific passwords for the iTC transporter

### PR DESCRIPTION
**Pull request by KrauseFx**
Taken from https://github.com/fastlane/fastlane/pull/4276

### CODE REVIEW

**Comment by KrauseFx** [comment]
The build will fail, since credentials_manager is not pushed yet

**Comment by asfalcone** [comment]
👍 LGTM

File: fastlane_core/lib/fastlane_core/itunes_transporter.rb [hunk 1]
```diff
@@ -262,11 +272,22 @@ def self.hide_transporter_output?
...
       use_shell_script ||= !ENV['FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT'].nil?
-      data = CredentialsManager::AccountManager.new(user: user, password: password)
 
+      # First, see if we have an application specific password
+      data = CredentialsManager::AccountManager.new(user: user,
+                                                  prefix: TWO_STEP_HOST_PREFIX)
       @user = data.user
-      @password = data.password
+      @password ||= data.password(ask_if_missing: false)
       @transporter_executor = use_shell_script ? ShellScriptTransporterExecutor.new : JavaTransporterExecutor.new
```
Comment (mfurtak) [inline fastlane_core/lib/fastlane_core/itunes_transporter.rb]: Can we move this line to before or after the password handling logic? It feels strange left in the middle here.

**Comment by mfurtak** [comment]
LGTM besides a small suggestion

File: fastlane_core/lib/fastlane_core/itunes_transporter.rb (hunk 1 above)
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/itunes_transporter.rb]: Very good point, I'l move it

**Comment by mfurtak** [comment]
:ship: when ready!

**Pull request merged**